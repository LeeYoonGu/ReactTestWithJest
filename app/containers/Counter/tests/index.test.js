/**
 * Test Counter
 */

import React from 'react';
import { shallow, mount } from 'enzyme';
import { browserHistory } from 'react-router-dom';
import { Provider } from 'react-redux';
import { fromJS } from 'immutable';
import configureStore from '../../../configureStore';
import { ConnectedRouter } from 'react-router-redux';
import createHistory from 'history/createBrowserHistory';

import { Counter, mapDispatchToProps } from '../index';
import {addNumber, subNumber} from "../actions";

import renderer from 'react-test-renderer'

const history = createHistory();
const store = configureStore({}, history);

describe('<Counter /> Test',()=>{

    let wrapper;
    let counter = 10;
    let store;
    beforeEach(()=>{
        wrapper = shallow(<Counter counter={counter}/>)
        store = configureStore({}, browserHistory);
    })

    it('<Counter /> Render', () => {
        expect(wrapper.length).toEqual(1)
    });

    it('<Counter /> Check Counter Number', () => {
        expect(wrapper.find('div[className="counter"]').props().children).toEqual(counter);
    });
});

describe('<ConnectedCounter /> ',()=>{
    const initialState = fromJS({
        counter: 1000,
    });
    let store,wrapper

    beforeEach(()=>{
        store = configureStore({}, browserHistory);
        wrapper = shallow(
            <Provider store={store}>
                <ConnectedCounter {...initialState}/>
            </Provider>
        )
    })
});

describe('Dispatch Actions ',()=> {
    it('AddNumber', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.onAddNumber();
        expect(dispatch).toHaveBeenCalledWith(addNumber());
    });

    it('SubNumber', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.onSubNumber();
        expect(dispatch).toHaveBeenCalledWith(subNumber());
    });
});

describe('Snapshot',()=>{
    it('+++capturing Snapshot of Home', () => {
        const renderedValue =  renderer.create(
            <Provider store={store}>
                <ConnectedRouter history={history}>
                    <Counter counter={100}/>
                </ConnectedRouter>
            </Provider>
        ).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
});