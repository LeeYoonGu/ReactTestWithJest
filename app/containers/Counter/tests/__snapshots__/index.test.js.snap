// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot +++capturing Snapshot of Home 1`] = `
<div>
  <div
    className="backButton"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <i
        className="fa fa-arrow-left fa-3x"
      />
    </a>
  </div>
  <div
    className="counter"
  >
    100
  </div>
  <div
    className="btnGroup"
  >
    <button
      className="btn"
      onClick={undefined}
    >
      <i
        className="fa fa-plus"
      />
    </button>
    <button
      className="btn"
      onClick={undefined}
    >
      <i
        className="fa fa-minus"
      />
    </button>
  </div>
</div>
`;
